4_params_munge/src/munge_bathy.R

```diff
@@ -101,3 +101,81 @@ munge_mndow_bathy <- function(out_ind, bathy_zip_ind, mndow_xwalk_ind){
     saveRDS(data_file)
   gd_put(out_ind, data_file)
 }
+
+munge_ndgf_bathy <- function(out_ind, ndgf_contour_ind, ndgf_xwalk_ind, ndgf_surface_sf_ind, crs){
+
+  ndgf_xwalk <- readRDS(sc_retrieve(ndgf_xwalk_ind))
+
+  # Now crosswalk with NHD lake ids
+  ndgf_contours <- readRDS(sc_retrieve(ndgf_contour_ind)) %>%
+    mutate(NDGF_ID = sprintf("ndgf_%s", LAKE)) %>%
+    inner_join(ndgf_xwalk, by = 'NDGF_ID') %>%
+    rename(other_ID = NDGF_ID,
+           depths = CONTOUR) %>%
+    dplyr::select(site_id, other_ID, depths)
+
+  # Due to some issues with incomplete surface polygons in the contour data
+  #   for some lakes causing zero area lakes and ultimately monotonicity issues,
+  #   replace all contour shapes (depth = 0) with the polygons from ndgf_lakesvw.
+
+  # We will treat these separately until after issues with some of the contour data are resolved
+
+  # Many of the lakes have multiple polygons at the same depth. Some of the smaller
+  # polygons at the same depth are contained within a larger polygon at the same depth.
+  # This doesn't make much sense and we end up double counting the area. So, union them
+  # first to reconcile internal boundaries by lake and then find the area.
+
+  ndgf_non_surface_geoms <- ndgf_contours %>%
+    filter(depths != 0) %>%
+    group_by(site_id, other_ID, depths) %>%
+    summarize(geometry = st_union(geometry)) %>%
+    mutate(areas = as.numeric(st_area(st_polygonize(geometry)))) %>%
+    dplyr::select(site_id, other_ID, depths, areas) %>%
+    ungroup() %>%
+    st_transform(crs = crs)
+
+  # Now isolate the surface geometries for the lakes in the contour dataset
+  # Make sure to join the crosswalk
+  ndgf_surface_geoms <- readRDS(sc_retrieve(ndgf_surface_sf_ind)) %>%
+    rename(NDGF_ID = site_id) %>%
+    left_join(ndgf_xwalk) %>%
+    filter(NDGF_ID %in% unique(ndgf_non_surface_geoms$other_ID)) %>%
+    mutate(depths = 0, areas = as.numeric(st_area(geometry))) %>%
+    # Needs to be in the same order as non surface one because we are binding rows
+    dplyr::select(site_id, other_ID = NDGF_ID, depths, areas) %>%
+    # Convert geometry type so it matches the type coming from contour data
+    st_cast("MULTILINESTRING") %>%
```

Comment:
why linestring and not multipolygon here?

Reply:
`MULTILINESTRING` because that is what `'3_params_fetch/out/ndgf_lake_contours.rds'` was read in as when I loaded the original shapefile. Was making sure I matched that in order to join.